File: server/php/src/Infrastructure/Mappers/PaymentDetails/ProtectionEligibilityMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\ProtectionEligibility as ProtectionEligibilityDto;
use OnlinePayments\Sdk\Domain\ProtectionEligibility as ProtectionEligibilitySdk;

final class ProtectionEligibilityMapper
{
    public static function mapFromSdkResponse(?ProtectionEligibilitySdk $response): ?ProtectionEligibilityDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new ProtectionEligibilityDto();
        $dto->eligibility = $response?->eligibility ?? null;
        $dto->type = $response?->type ?? null;

        return $dto;
    }
}

File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/ProtectionEligibilityMapper.cs
using ProtectionEligibilitySdk = OnlinePayments.Sdk.Domain.ProtectionEligibility;
using ProtectionEligibilityDto = Business.Domain.Payments.PaymentDetails.ProtectionEligibility;

namespace Infrastructure.Mappers.PaymentDetails;

public static class ProtectionEligibilityMapper
{
    public static ProtectionEligibilityDto Map(ProtectionEligibilitySdk? response)
    {
        return new()
        {
            Eligibility = response?.Eligibility,
            Type = response?.Type
        };
    }
}